File: rockpaperscissors.py
from random import choice


def tieptuc(temp):
    match temp:
        case 'y' | 'Y':
            print("You chose to continue game.")
            main()
        case 'n' | 'N':
            print("Thanks for playing!")
        case _:
            print("Error, just type 'y' or 'n' only!")
            return tieptuc(str(input("Do you wanna continue?(y/n) ")))


def main():
    player = str(input("Enter [rock], [paper], [scissors]: "))
    choose = ['rock', 'paper', 'scissors']
    bot = choice(choose)

    print(f"You chose: {player}\nBot chose: {bot}")

    if player == bot:
        print("Tie!")
        tieptuc(str(input("Do you wanna continue?(y/n) ")))
    elif player == 'rock':
        if bot == 'scissors':
            print("You win!")
            tieptuc(str(input("Do you wanna continue?(y/n) ")))
        else:
            print("You lose!")
            tieptuc(str(input("Do you wanna continue?(y/n) ")))
    elif player == 'paper':
        if bot == 'rock':
            print("You win!")
            tieptuc(str(input("Do you wanna continue?(y/n) ")))
        else:
            print("You lose!")
            tieptuc(str(input("Do you wanna continue?(y/n) ")))
    else:
        if bot == 'paper':
            print("You win!")
            tieptuc(str(input("Do you wanna continue?(y/n) ")))
        else:
            print("You lose!")
            tieptuc(str(input("Do you wanna continue?(y/n) ")))


main()

File: rockpaperscissors.cpp
#include <iostream>
#include <string>
#include <time.h>
#include <stdlib.h>
using namespace std;

bool continue_game = true;

void tieptuc()
{
    std::cout << "Do you wanna continue?(y/n) ";
    char temp;
    std::cin >> temp;
    switch (temp)
    {
    case 'y':
    case 'Y':
        std::cout << "You chose to continue.\n";
        break;
    case 'n':
    case 'N':
        std::cout << "Thanks for playing!\n";
        continue_game = false;
        break;
    default:
        std::cout << "Error, just type 'y' or 'n' only!\n";
        tieptuc();
        break;
    }
}

int main()

{
    std::string player;
    std::string bot;
    std::string types[3] = {
        "rock",
        "paper",
        "scissors"
    };

    while (continue_game)
    {
        std::cout << "Enter [rock], [paper], [scissors]: ";
        std::cin >> player;

        srand(time(NULL));
        int randomnumber;
        randomnumber = rand() % 3;
        bot = (types[randomnumber]);

        std::cout << "You chose: " << player << "\nBot chose: " << bot << std::endl;

        if (player == bot)
        {
            std::cout << "Tie!\n";
            tieptuc();
        }
        else if (player == "rock")
        {
            if (bot == "scissors")
            {
                std::cout << "You win!\n";
                tieptuc();
            }
            else
            {
                std::cout << "You lose!\n";
                tieptuc();
            }
        }
        else if (player == "paper")
        {
            if (bot == "rock")
            {
                std::cout << "You win!\n";
                tieptuc();
            }
            else{
                std::cout << "You lose!\n";
                tieptuc();
            }
        }
        else if (player == "scissors")
        {
            if (bot == "paper")
            {
                std::cout << "You win!\n";
                tieptuc();
            }
            else
            {
                std::cout << "You lose!\n";
                tieptuc();
            }
        }
    }
    return 0;
}